src/main/java/graphql/nadel/engine/transformation/RecordTypeInformation.java

```diff
@@ -0,0 +1,131 @@
+package graphql.nadel.engine.transformation;
+
+import graphql.language.AstNodeAdapter;
+import graphql.language.Field;
+import graphql.language.FragmentDefinition;
+import graphql.language.InlineFragment;
+import graphql.language.Node;
+import graphql.language.NodeVisitorStub;
+import graphql.nadel.engine.FieldMetadataUtil;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLFieldsContainer;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLSchema;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+import graphql.util.TraverserVisitorStub;
+import graphql.util.TreeTransformer;
+import graphql.util.TreeTransformerUtil;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static graphql.Assert.assertNotNull;
+import static graphql.introspection.Introspection.TypeNameMetaFieldDef;
+import static graphql.schema.GraphQLTypeUtil.unwrapAll;
+
+public class RecordTypeInformation {
```

Comment:
Should these classes go into their own top level classes.

There a fair amount of code here

Reply:
done

Reply:
Also you know me - I love at least a class JavaDoc comment on what it does